.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,100 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '16 12 * * 6'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    env:
+      COMPILED: |
+        [
+          "cpp"
+        ]
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'java', 'javascript', 'python', 'ruby' ]
```

Comment:
Not sure it makes sense to include ruby here, as we have no ruby code (except the test sources).